index.d.ts

```diff
@@ -0,0 +1,104 @@
+import stream = require('stream');
+
+declare function request(superagent: any, promise: any): request.SuperAgentStatic;
+
+declare namespace request {
+  interface SuperAgentStatic extends SuperAgent<SuperAgentRequest> {
+    (url: string): SuperAgentRequest;
+    (method: string, url: string): SuperAgentRequest;
+
+    agent(): SuperAgent<SuperAgentRequest>;
```

Comment:
These signatures are only available on the superagent together with `get()` etc, correct? I think these should just be moved to `SuperAgent` and this interface can be removed